### PR TITLE
Address- EOS-20437: Configure Network and EOS-20438: Factory Node Initialize

### PR DESCRIPTION
**Pull request by SomwanshiAnjali**
1. Add lr-cli directory
2. create cortx_setup package with initial config in setup.py
3. Add wrapper for `configure network` which can be invoked through `nodecli` 

Signed-off-by: Anjali Somwanshi <anjali.somwanshi@seagate.com>

### CODE REVIEW

File: lr-cli/cortx_setup/network.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    """Configure network"""
+
+    def config(self, transport_type=None, interface_type=None,
+            network_type=None, interfaces=None
+    ):
+        """Update provided network details in pillar data"""
+
+        node_id = local_minion_id()
+        if transport_type is not None:
+            logger.info("Updating transport type in pillar data")
```
Comment (ypise) [inline lr-cli/cortx_setup/network.py:21]: Update to:
```logger.info(f"Updating transport type to {transport_type} in pillar data")```

File: lr-cli/cortx_setup/network.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            PillarSet().run(
+                f'node_info/{node_id}/network/data/transport_type',
+                f'{transport_type}',
+                targets=node_id,
+                local=True
+            )
+
+        if interface_type is not None:
+            logger.info("Updating interface type in pillar data")
```
Comment (ypise) [inline lr-cli/cortx_setup/network.py:30]: Update this to:
```logger.info("Updating interface type to {interface_type} in pillar data")```

File: lr-cli/cortx_setup/network.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+
+class Network():
+
+    """Configure network"""
+
+    def config(self, transport_type=None, interface_type=None,
+            network_type=None, interfaces=None
+    ):
+        """Update provided network details in pillar data"""
```
Comment (ypise) [inline lr-cli/cortx_setup/network.py:17]: Provide sample values for the parameters in doc string.

File: lr-cli/cortx_setup/network.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+            )
+
+        if network_type is not None:
+            if interfaces is not None:
+                logger.info(
+                    f"Updating interfaces for {network_type} network "
+                    "in pillar data"
+                )
+                PillarSet().run(
+                    f'node_info/{node_id}/network/{network_type}/intefaces',
```
Comment (ypise) [inline lr-cli/cortx_setup/network.py:45]: Do we need interface type here? (interface_type: public/private)
Network type is Mgmt/Data, so we need  a parameter to capture the interface type.
In this context, interface type has to be implicit parameter inside network.

File: lr-cli/cortx_setup/network.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+                )
+            else:
+                logger.error(
+                    f"Interfaces should be provided for {network_type} network"
+                )
+        logger.info("Done")
+
+
+def _parse_args():
+    parser = argparse.ArgumentParser(description='Configure netowrk')
```
Comment (ypise) [inline lr-cli/cortx_setup/network.py:58]: Configure netowrk => Configure network

File: lr-cli/cortx_setup/network.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    parser.add_argument('--transport_type', type=str,
```
Comment (ypise) [inline lr-cli/cortx_setup/network.py:59]: Provide sample data for arguments.

File: lr-cli/cortx_setup/node.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
+import argparse
+import logging
+
+from provisioner.commands import deploy
+from provisioner.salt import local_minion_id
+
+logger = logging.getLogger(__name__)
+
+cortx_components = dict(
```
Comment (ypise) [inline lr-cli/cortx_setup/node.py:9]: Are we clubbing 2 tickets in 1 PR?
If so, update the PR title accordingly.

**Review by ypise — CHANGES_REQUESTED** [review]
Few suggestions to be addressed.

File: lr-cli/cortx_setup/network.py (hunk 1 above)
Comment (SomwanshiAnjali) [reply lr-cli/cortx_setup/network.py]: Addressed

File: lr-cli/cortx_setup/network.py (hunk 2 above)
Comment (SomwanshiAnjali) [reply lr-cli/cortx_setup/network.py]: Addressed

File: lr-cli/cortx_setup/network.py (hunk 3 above)
Comment (SomwanshiAnjali) [reply lr-cli/cortx_setup/network.py]: Addressed

File: lr-cli/cortx_setup/network.py (hunk 4 above)
Comment (SomwanshiAnjali) [reply lr-cli/cortx_setup/network.py]: Addressed

File: lr-cli/cortx_setup/network.py (hunk 5 above)
Comment (SomwanshiAnjali) [reply lr-cli/cortx_setup/network.py]: Addressed

File: lr-cli/cortx_setup/network.py (hunk 6 above)
Comment (SomwanshiAnjali) [reply lr-cli/cortx_setup/network.py:93]: Addressed

**Review by ypise — APPROVED** [review]

**Pull request merged**